archive/zip_archiver.go

```diff
@@ -47,7 +47,25 @@ func (a *ZipArchiver) ArchiveFile(infilename string) error {
 		return err
 	}
 
-	return a.ArchiveContent(content, fi.Name())
+	if err := a.open(); err != nil {
+		return err
+	}
+	defer a.close()
+
+	fh, err := zip.FileInfoHeader(fi)
+	if err != nil {
+		return fmt.Errorf("error creating file header: %s", err)
+	}
+	fh.Name = fi.Name()
+	fh.Method = zip.Deflate
+
+	f, err := a.writer.CreateHeader(fh)
+	if err != nil {
+		return fmt.Errorf("error creating file inside archive: %s", err)
+	}
+
```

Comment:
do you need to defer an `f.Close` here?

Reply:
I'm not sure. It's not done in `ArchiveDir`:
https://github.com/terraform-providers/terraform-provider-archive/blob/0e04dbb7513bd0151028fabe05a871cb9214cbf4/archive/zip_archiver.go#L80

So if we need it then I should add it to both? Let me know and I'll add it.

Reply:
Getting this error if I try to add it:

```
$ go build
# github.com/terraform-providers/terraform-provider-archive/archive
../../go/src/github.com/terraform-providers/terraform-provider-archive/archive/zip_archiver.go:66:9: f.close undefined (type io.Writer has no field or method close)
```

Reply:
Sorry this was my mistake, i thought it was an `io.WriteCloser` or something, not just an `io.Writer`.